### PR TITLE
player becomes dead upon reconnect, after 30s disconnect

### PR DESCRIPTION
**Pull request by nevinlazarus**
Sets a date field on the server when client disconnects
If more that 30s has elapsed, upon reconnect the player becomes dead

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -82,6 +87,15 @@ io.on('connection', function (socket) {
     // when the user disconnects.. perform this
     socket.on('disconnect', function () {
         --numUsers;
+        
+        //gets current time
+        var d = new Date();
+        //sets dead timeout for 30 seconds
+        if (user[socket.username]) {
+            disconnect.setTime(d.getTime() + 30 * 1000);
```
Comment (vincentt143-zz) [inline server.js]: Chuck the timeout in a global variable at the top of the file. Other than that, looks good to merge.

File: server.js [hunk 2]
```diff
@@ -19,6 +19,9 @@ var users = {};
 var numUsers = 0;
 var status = ("surv", "zomb", "dead");
 
+//30s disconnect timeout
+var dcTimeout = 30;
```
Comment (vincentt143-zz) [inline server.js]: Use capitalisation (DISCONNECT_TIMEOUT) for constants. There is a const modifier in javascript but some browsers don't support it.

File: server.js [hunk 3]
```diff
@@ -43,7 +46,8 @@ io.on('connection', function (socket) {
             users[username] = {
                 latitude: 0,
                 longitude: 0,
-                stat: status[Math.random(1)]
+                stat: status[Math.random(1)],
```
Comment (vincentt143-zz) [inline server.js]: Have a look at Mendel's branch and set it to whatever survivor is by default.

**Comment by vincentt143-zz** [comment]
A few small changes but it should be good to merge.

**Comment by nevinlazarus** [comment]
fixed, should I merge now?

**Comment by vincentt143-zz** [comment]
:+1:

**Pull request merged**